keras_cv/layers/object_detection/non_max_suppression.py

```diff
@@ -121,12 +121,18 @@ def call(self, predictions, images=None):
             self.confidence_threshold,
             clip_boxes=False,
         )
-
         # output will be a ragged tensor because num_boxes will change across the batch
-        return self._encode_to_ragged(nmsed_boxes, images)
+        boxes = self._encode_to_dense_output_tensor(nmsed_boxes)
+        # converting all boxes to the original format
+        boxes = bounding_box.convert_format(
```

Comment:
Since images are batched, the boxes need to be batched as well. This small fix may help:

```python
        boxes = bounding_box.convert_format(
            tf.expand_dims(boxes, axis = 0),
            source="yxyx",
            target=self.bounding_box_format,
            images=images,
        )[0]
```

Reply:
Moved to be done after ragged encoding.